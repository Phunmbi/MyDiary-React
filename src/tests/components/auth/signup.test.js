import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import SignupComponent, { Signup } from '../../../components/auth/Signup';
import Root from '../../../root';

let wrapper;
let wrapped;
let wrappedTrue;

const response = {
  auth: {
    error: false,
  },
  addEntry: jest.fn(),
  entries: {
    newEntryError: {
      data: {
        message: 'new',
      },
      length: 5,
    },
  },
};

const history = {
  push: jest.fn(),
};

const responseTrue = {
  auth: {
    status: 201,
    error: {
      data: {
        Error: true,
      },
    },
  },
  addEntry: jest.fn(),
  entries: {
    newEntryError: {
      data: {
        message: 'new',
      },
      length: 5,
    },
  },
};

document.getElementById = jest.fn((selector) => {
  console.log('selector => ', selector);

  return {
    style: {
      background: 'background',
    },
    innerHTML: 'inner',
    disabled: true,
  };
});

beforeEach(() => {
  jest.useFakeTimers();
  wrapper = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
        <SignupComponent
          response={response}
        />
      </MemoryRouter>
    </Root>,
  );

  wrapped = shallow(
    <Signup
      response={response}
    />,
  );


  wrappedTrue = shallow(
    <Signup
      response={responseTrue}
      history={history}
    />,
  );
});

afterEach(() => wrapper.unmount());

describe('Signup UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Signup Functionality', () => {
  it('should click simulate submit', () => {
    const submitButton = wrapped.find('form');
    const firstName = wrapped.find('#firstName');

    const mockedEvent = {
      preventDefault: jest.fn(),
      target: {
        reset: jest.fn(),
        children: {
          firstName: {
            value: 'value',
          },
          lastName: {
            value: 'value',
          },
          email: {
            value: 'value',
          },
          password: {
            value: 'value',
          },
          confirmPassword: {
            value: 'value',
          },
          title: 'title',
          details: 'details',
          submit: {
            innerHTML: 'submit',
            style: {
              background: 'background',
            },
          },
        },
      },
    };

    submitButton.simulate('submit', mockedEvent);
    firstName.simulate('change', mockedEvent);
  });

  it('should click simulate submit', () => {
    const submitButton = wrappedTrue.find('form');
    const firstName = wrappedTrue.find('#firstName');

    const mockedEvent = {
      preventDefault: jest.fn(),
      target: {
        reset: jest.fn(),
        children: {
          firstName: {
            value: 'value',
          },
          lastName: {
            value: 'value',
          },
          email: {
            value: 'value',
          },
          password: {
            value: 'value',
          },
          confirmPassword: {
            value: 'value',
          },
          title: 'title',
          details: 'details',
          submit: {
            innerHTML: 'submit',
            style: {
              background: 'background',
            },
          },
        },
      },
    };

    submitButton.simulate('submit', mockedEvent);
    firstName.simulate('change', mockedEvent);
  });
});
