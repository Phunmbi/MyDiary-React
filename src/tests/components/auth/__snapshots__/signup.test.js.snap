// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Signup)
          response={
            Object {
              "addEntry": [MockFunction],
              "auth": Object {
                "error": false,
              },
              "entries": Object {
                "newEntryError": Object {
                  "data": Object {
                    "message": "new",
                  },
                  "length": 5,
                },
              },
            }
          }
        >
          <Signup
            response={
              Object {
                "auth": Object {
                  "authenticated": null,
                  "firstName": null,
                  "lastName": null,
                },
                "entries": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
                "profile": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
              }
            }
            signup={[Function]}
          >
            <div
              className="hero-account"
            >
              <h3>
                Create your MyDiary Account
              </h3>
              <p
                className="errorResponse"
                id="errorResponse"
              />
              <form
                onSubmit={[Function]}
              >
                <input
                  id="firstName"
                  onChange={[Function]}
                  placeholder="First Name"
                  required={true}
                  type="text"
                />
                <span
                  className="errorResponse"
                />
                <input
                  id="lastName"
                  onChange={[Function]}
                  placeholder="Last Name"
                  required={true}
                  type="text"
                />
                <span
                  className="errorResponse"
                />
                <input
                  id="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                />
                <span
                  className="errorResponse"
                />
                <input
                  id="password"
                  min="8"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />
                <span
                  className="errorResponse"
                />
                <input
                  id="confirmPassword"
                  min="8"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required={true}
                  type="password"
                />
                <span
                  className="errorResponse"
                />
                <button
                  id="submit"
                  type="submit"
                >
                  Create Account
                </button>
              </form>
              <p>
                Already have an account? then
                <Link
                  replace={false}
                  to="/auth/signin"
                >
                  <a
                    href="/auth/signin"
                    onClick={[Function]}
                  >
                     Sign in
                  </a>
                </Link>
              </p>
            </div>
          </Signup>
        </Connect(Signup)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
